MantidPlot/src/Mantid/MantidMatrix.cpp

```diff
@@ -1326,17 +1332,24 @@ void MantidMatrix::setupNewExtension(MantidMatrixModel::Type type) {
 void MantidMatrix::updateExtensions(Mantid::API::MatrixWorkspace_sptr ws) {
   auto it = m_extensions.begin();
   while (it != m_extensions.cend()) {
-    switch (it->first) {
+    auto type = it->first;
+    switch (type) {
     case MantidMatrixModel::DX:
       if (ws->hasDx(0)) {
         auto &extension = it->second;
         extension.model = new MantidMatrixModel(this, ws.get(), m_rows, m_cols,
-                                                m_startRow, it->first);
+                                                m_startRow, type);
         connectTableView(extension.tableView.get(), extension.model);
+        auto format = m_extensionRequest.getFormat(
+            type, m_extensions, MantidPreferences::MantidMatrixNumberFormatY());
+        auto precision = m_extensionRequest.getPrecision(
+            type, m_extensions,
+            MantidPreferences::MantidMatrixNumberPrecisionY());
```

Comment:
Should these be `MantidMatrixNumberPrecisionDx` and `MantidMatrixNumberFormatDx` instead?

Reply:
Yes, this was copied from the code in setupNewExtension, which should also have been using the `Dx` versions!